dokka-subprojects/plugin-base/src/main/kotlin/org/jetbrains/dokka/base/translators/documentables/DefaultPageCreator.kt

```diff
@@ -380,73 +388,58 @@ public open class DefaultPageCreator(
-        ).flatten()
-        return contentForScope(setOf(dri), sourceSets, types, s.functions, s.properties, extensions)
-    }
+    ): ContentGroup = contentForScopes(listOf(s), sourceSets, extensions)
 
     private fun contentForScope(
         dri: Set<DRI>,
         sourceSets: Set<DokkaSourceSet>,
         types: List<Documentable>,
         functions: List<DFunction>,
         properties: List<DProperty>,
-        extensions: List<Documentable>
+        extensions: List<Documentable>,
     ) = contentBuilder.contentFor(dri, sourceSets) {
-        divergentBlock(
-            "Types",
-            types,
-            ContentKind.Classlikes
-        )
-        val (extensionProps, extensionFuns) = extensions.splitPropsAndFuns()
+        typesBlock(types)
+        val (extensionProperties, extensionFunctions) = extensions.splitPropsAndFuns()
         if (separateInheritedMembers) {
             val (inheritedFunctions, memberFunctions) = functions.splitInherited()
             val (inheritedProperties, memberProperties) = properties.splitInherited()
 
-            val (inheritedExtensionFunctions, extensionFunctions) = extensionFuns.splitInheritedExtension(dri)
-            val (inheritedExtensionProperties, extensionProperties) = extensionProps.splitInheritedExtension(dri)
-            propertiesBlock(
-                "Properties", memberProperties + extensionProperties
-            )
-            propertiesBlock(
-                "Inherited properties", inheritedProperties + inheritedExtensionProperties
-            )
-            functionsBlock("Functions", memberFunctions + extensionFunctions)
-            functionsBlock(
-                "Inherited functions", inheritedFunctions + inheritedExtensionFunctions
-            )
+            val (
+                inheritedExtensionFunctions,
+                directExtensionFunctions
+            ) = extensionFunctions.splitInheritedExtension(dri)
+
+            val (
+                inheritedExtensionProperties,
+                directExtensionProperties
+            ) = extensionProperties.splitInheritedExtension(dri)
+
+            propertiesBlock("Properties", memberProperties, directExtensionProperties)
+            propertiesBlock("Inherited properties", inheritedProperties, inheritedExtensionProperties)
+
+            functionsBlock("Functions", memberFunctions, directExtensionFunctions)
+            functionsBlock("Inherited functions", inheritedFunctions, inheritedExtensionFunctions)
         } else {
-            propertiesBlock(
-                "Properties", properties + extensionProps
-            )
-            functionsBlock("Functions", functions + extensionFuns)
+            propertiesBlock("Properties", properties, extensionProperties)
+            functionsBlock("Functions", functions, extensionFunctions)
         }
     }
 
-    private fun Iterable<DFunction>.sorted() =
-        sortedWith(compareBy({ it.name }, { it.parameters.size }, { it.dri.toString() }))
```

Comment:
Hmm, did this sort get lost accidentally? I wonder if #1333 became relevant again, I don't see any tests in the PR that fixed it

Reply:
Not accidentally but because of its redundancy. After https://github.com/Kotlin/dokka/pull/2846 sorting is now performed inside divergent group - specifically [here](https://github.com/Kotlin/dokka/blob/2bcd3da3fb2726fd56a316bb9d78ee4faa5b5b16/dokka-subprojects/plugin-base/src/main/kotlin/org/jetbrains/dokka/base/translators/documentables/DefaultPageCreator.kt#L859C34-L859C34)

that's why no tests changed
but, I will recheck original issue #1333, just in case

Reply:
Checked this case — even with shuffling of members, the order in documentation stays stable.